hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -938,12 +980,38 @@ private void compact(String compactionCommitTime) throws IOException {
    * Performs a compaction operation on a dataset. WARNING: Compaction operation cannot be executed
    * asynchronously. Please always use this serially before or after an insert/upsert action.
    */
-  public String forceCompact() throws IOException {
-    String compactionCommitTime = HoodieActiveTimeline.createNewCommitTime();
-    compact(compactionCommitTime);
+  private String forceCompact() throws IOException {
+    String compactionCommitTime = startCompaction();
+    forceCompact(compactionCommitTime);
     return compactionCommitTime;
   }
 
+  private void commitCompaction(JavaRDD<WriteStatus> writeStatuses,
```

Comment:
This method is required because inline compaction is called with commit(...) and so cannot call commit again inside inline compaction(..)

Reply:
rename to `commitForceCompaction` , to denote intent to use it with forced compactions?

Reply:
yeah, good point, done